### PR TITLE
Implemented timers

### PR DESCRIPTION
**Pull request by Subv**
This PR implements the Timer kernel object.
The timer starts counting when SetTimer is called, and it awakes the waiting threads when the specified interval has passed.

Based on top of #425
Also requires #439

### CODE REVIEW

**Comment by Subv** [comment]
This will probably require the implementation of the CoreTiming namespace from Dolphin / PPSSPP. I have an idea on how to implement the timeouts and waits but we'll be needing that first.

**Comment by Subv** [comment]
Updated this PR to base it on #425, should be ready now, just waiting on #425 and the fix for the idle thread issue.

File: src/core/hle/kernel/timer.cpp [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+    timer->signaled = false;
+    return RESULT_SUCCESS;
+}
+
+/// The event type of the generic timer callback event
+static int TimerCallbackEventType = -1;
+
+/// The timer callback event, called when a timer is fired
+static void TimerCallback(u64 timer_handle, int cyclesLate) {
```
Comment (bunnei) [inline src/core/hle/kernel/timer.cpp]: Variable naming of `cyclesLate`

File: src/core/hle/kernel/timer.cpp [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    Timer* timer = Kernel::g_handle_table.Get<Timer>(handle);
+
+    if (timer == nullptr)
+        return InvalidHandle(ErrorModule::Kernel);
+
+    timer->initial_delay = initial;
+    timer->interval_delay = interval;
+
+    u64 initial_microseconds = initial / 1000;
+    CoreTiming::ScheduleEvent(usToCycles(initial_microseconds), TimerCallbackEventType, handle);
```
Comment (bunnei) [inline src/core/hle/kernel/timer.cpp]: Not really relevant to this PR, but I wish we fixed the naming of `usToCycles`, I must have missed it before

File: src/core/hle/kernel/timer.cpp [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    if (timer->interval_delay != 0) {
+        // Reschedule the timer with the interval delay
+        u64 interval_microseconds = timer->interval_delay / 1000;
+        CoreTiming::ScheduleEvent(usToCycles(interval_microseconds) - cyclesLate, 
+                TimerCallbackEventType, timer_handle);
+    }
+}
+
+ResultCode SetTimer(Handle handle, s64 initial, s64 interval) {
+    Timer* timer = Kernel::g_handle_table.Get<Timer>(handle);
```
Comment (bunnei) [inline src/core/hle/kernel/timer.cpp]: If initial/interval are signed, is that because -1 means something?

Comment (Subv) [inline src/core/hle/kernel/timer.cpp]: I'm just following the signature from 3dbrew, I don't think -1 is a valid value but i'm not sure

Comment (bunnei) [inline src/core/hle/kernel/timer.cpp]: Should we check that?

Comment (Subv) [inline src/core/hle/kernel/timer.cpp]: We should probably do a hwtest about it, if it's valid then the timer will never fire (and this doesn't make any sense)

**Comment by bunnei** [comment]
LGTM

**Pull request merged**